support "events" in general and "cuepoint" event in particular

skaurus opened the pull request:
cuepoint event is the same as in flowplayer: https://flowplayer.org/docs/cuepoints.html

skaurus committed:
support events in general and cuepoint event in particular

cuepoint event is the save as in flowplayer: https://flowplayer.org/docs/cuepoints.html

skaurus committed:
Merge pull request #1 from skaurus/skaurus-cuepoints

support "events" in general and "cuepoint" event in particular

skaurus committed:
fix

skaurus committed:
"ready" event which provide info about video width, height and duration

skaurus committed:
added setCuepoints method

now we can add cuepoints on "ready" event and subscribe to "cuepoint" event like this:

`player = new CanvasVideoPlayer({
    ...
}).on('ready', function (data) {
    console.log('ready', data)
    var cuepoints = [];
    for (var i = 1; i <= 9; i++) {
        cuepoints.push( data.duration * i / 10 );
    }
    this.setCuepoints(cuepoints);
}).on('cuepoint', function (data) {
    console.log(new Date(), data);
});`

skaurus committed:
added "progress" and "finish" "events"

skaurus commented:
This is not real events at all, but at least for me they will do. I feel that this will be true for others as well.

Stanko commented:
Thanks a lot, I will merge it for sure. Would you mind, adding a few lines about it to the documentation?

skaurus commented:
I'm not sure I will have time for this in next few days :(

Stanko commented:
No worries, take your time. Also formatting and coding conventions are bit out of sync.

skaurus committed:
document "events", new option and new method

skaurus commented:
I have updated documentation.
Not sure what you mean by formatting and conventions problems - I even fixed two your formatting bugs :) Please fix these problems yourself as you see fit.

Commit:
Merge remote-tracking branch 'upstream/gh-pages' into gh-pages

Commit:
merge

skaurus committed:
fixing mulitple firing 'finish' event while scrolling

For some reason condition which hosts 'finish' event is fired multiple times when page with stopped video is scrolled.

sethbrandom commented:
Hi, Mr. Stanko. I really appreciate any help... I already looking many differents ways to implement a video player on iPhone that not trigger fullscreen, but your implementation seems to have the exact what I need. But I need in my project config pauses along side the video timeline, and the user click on a next button event to resume playing (or previous to return to "previous" mark on this config), but I'm not sure how to do this... I used to work with the videojs player, because this already has a plugin to do so (insert chapters marks along the video), but videojs not avoid iPhone open fullscreen. Please, any help I'll be very glad. 

Thanks 

Seth
